### PR TITLE
Replace unmaintained `apache-httpd` project with new one

### PR DESCRIPTION
**Pull request by yakatz**
(apache-httpd) (apachelounge-httpd) Replace unmaintained Apache HTTPD package

## Description
Create a new package based on the ApacheLounge HTTPD build to replace the Apache Haus build which is no longer maintained. The package name has been changed to reflect that the build has changed, but they are supposed to be binary-compatible.

## Motivation and Context
The old package is no longer updated.

Related discussion: https://github.com/orgs/chocolatey-community/discussions/2216

## How Has this Been Tested?
Installed manually on Windows 10 and Windows 11.  
I was not able to get the `chocolatey-test-environment` running on my machine - possibly a Vagrant issue

## Checklist:

- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [x] I have updated the package description and it is less than 4000 characters.
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md)
- [ ] The added/modified package passed install/uninstall in the Chocolatey Test Environment(https://github.com/chocolatey-community/chocolatey-test-environment/). _Note that we don't support the use of any other environment_.
- [ ] The changes only affect a single package (not including meta package).

### CODE REVIEW

**Comment by yakatz** [comment]
Replaces #2381 and #2321

File: automatic/apachelounge-httpd/tools/helpers.ps1 [hunk 1]
```diff
@@ -97,9 +102,12 @@ function Install-ApacheService {
 
     $apachePaths = Get-ApachePaths $arguments.destination
 
+    Write-Warning "This command is expected to write debugging output. Chocolately treats that as an error. Please read the error text to confirm whether service installation succeed."
     & $apachePaths.BinPath -k install -n "$($arguments.serviceName)"
```
Comment (JPRuskin) [inline automatic/apachelounge-httpd/tools/helpers.ps1:106]: Would something like this fix this up? Given that it does raise things that need fixing, I'd almost be inclined to suggest outputting it as a warning if there was anything found, but that's more processing...

```suggestion
    & $apachePaths.BinPath -k install -n "$($arguments.serviceName)" 2>&1
```

I believe that should work back to 2.0.

Comment (JPRuskin) [reply automatic/apachelounge-httpd/tools/helpers.ps1:106]: Example output from a quick test:
image

Comment (JPRuskin) [reply automatic/apachelounge-httpd/tools/helpers.ps1:106]: Example of highlighting the errors:

```suggestion
    # This command outputs to the error stream. 2>&1 redirects that to output.
    & $apachePaths.BinPath -k install -n "$($arguments.serviceName)" 2>&1 | ForEach-Object {
        if ($_ -match "^\w+\d+: ") {
            Write-Warning "$_"
        }
        else {
            Write-Host "$_"
        }
    }
```

image

Comment (JPRuskin) [reply automatic/apachelounge-httpd/tools/helpers.ps1:106]: @AdmiringWorm Any thoughts on my pushing the changes suggested in the last comment? I think it makes sense, then we can merge this in.

Comment (AdmiringWorm) [reply automatic/apachelounge-httpd/tools/helpers.ps1:106]: Let us not go with any custom parsing inside the package, so I would say just go with the first option. We can always make changes at a later date if needed.

Comment (JPRuskin) [reply automatic/apachelounge-httpd/tools/helpers.ps1:106]: Ah, that's a shame, as it seemed useful - but fair enough, it's a line. I'll pop that in.

File: automatic/apachelounge-httpd/tools/chocolateyInstall.ps1 [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    file             = "$toolsDir\httpd-2.4.59-240404-win32-vs17.zip"
+    file64           = "$toolsDir\httpd-2.4.59-240404-win64-VS17.zip"
+    destination      = if ($pp.installLocation) { $pp.installLocation } else { $env:APPDATA }
+    port             = if ($pp.Port) { $pp.Port } else { 8080 }
+    serviceName      = if ($pp.NoService) { $null } elseif ($pp.serviceName) { $pp.serviceName } else { 'Apache2.4' }
+    startService     = if ($pp.NoStartService) { $false } else { $true }
+    setConfiguration = if ($pp.SkipConfiguration) { $false } else { $true }
+}
+
+if ($arguements.setConfiguration -and -not (Assert-TcpPortIsOpen $arguments.port)) {
```
Comment (JPRuskin) [inline automatic/apachelounge-httpd/tools/chocolateyInstall.ps1:17]:
```suggestion
if ($arguments.setConfiguration -and -not (Assert-TcpPortIsOpen $arguments.port)) {
```

File: automatic/apachelounge-httpd/tools/helpers.ps1 [hunk 3]
```diff
@@ -97,9 +102,12 @@ function Install-ApacheService {
 
     $apachePaths = Get-ApachePaths $arguments.destination
 
-    & $apachePaths.BinPath -k install -n "$($arguments.serviceName)"
+    # This command outputs to the error stream. 2>&1 redirects that to output.
+    & $apachePaths.BinPath -k install -n "$($arguments.serviceName)" 2>&1
 
-    Start-Service $arguments.serviceName
+    if ($arguements.StartService) {
```
Comment (JPRuskin) [inline automatic/apachelounge-httpd/tools/helpers.ps1:108]:
```suggestion
    if ($arguments.StartService) {
```

File: automatic/apachelounge-httpd/tools/helpers.ps1 [hunk 4]
```diff
@@ -164,5 +172,6 @@ function Uninstall-ApacheService {
 
     $apachePaths = Get-ApachePaths $arguments.destination
 
+    Write-Warning "This command is expected to write debugging output. Chocolately treats that as an error. Please read the error text to confirm whether service installation succeed."
     & $apachePaths.BinPath -k uninstall -n "$($arguments.serviceName)"
```
Comment (JPRuskin) [inline automatic/apachelounge-httpd/tools/helpers.ps1:176]:
```suggestion
    # This command outputs to the error stream. 2>&1 redirects that to output.
    & $apachePaths.BinPath -k uninstall -n "$($arguments.serviceName)" 2>&1
```

Replicate the pattern from the other call similar to this.

**Pull request closed without merging**